SharedSources/MediaLibraryModel/MediaModel.swift

```diff
@@ -35,12 +35,34 @@ extension MediaModel {
 
 // MARK: - Helpers
 
+extension MediaModel {
+    /// Swap the given [VLCMLMedia] to the cached array.
+    /// This only swaps media with the same VLCMLIdentifiers
+    /// - Parameter medias: To be swapped medias
+    /// - Returns: New array of `VLCMLMedia` if changes have been made, else return a unchanged cached version.
+    func swapMedias(with medias: [VLCMLMedia]) -> [VLCMLMedia] {
+        var newFiles = files
+
+        // FIXME: This should be handled in a thread safe way
```

Comment:
I could directly integrate the thread safety, but ideally, I would want to go through the whole model/medialibrary integration. I would be for merging this without, and doing a follow-up ticket for this.